docs/rfcs/015-iiif-av-delivery-channel-settings.md

```diff
@@ -0,0 +1,132 @@
+
+AV policies are a comma separated list found in the `ImageOptimisationPolcies` table that can consist of either elastic transcoder standard presets and friendly names that link to values in appsettings.
+
+The system presets can be seen by using the aws command `aws elastictranscoder list-presets --query="Presets[].Name"` and this comes back like this:
+
+```json
+[
+    "System preset: Generic 1080p",
+    "System preset: Generic 720p",
+    "System preset: Generic 480p 16:9",
+    "System preset: Generic 480p 4:3",
+    "System preset: Generic 360p 16:9",
+    "System preset: Generic 360p 4:3",
+    "System preset: Generic 320x240",
+    "System preset: iPhone4S",
+    "System preset: iPod Touch",
+    "System preset: Apple TV 2G",
+    "System preset: Apple TV 3G",
+    "System preset: Web",
+    "System preset: KindleFireHD",
+    "System preset: KindleFireHD8.9",
+    "System preset: Audio AAC - 256k",
+    "System preset: Audio AAC - 160k",
+    "System preset: Audio AAC - 128k",
+    "System preset: Audio AAC - 64k",
+    "System preset: KindleFireHDX",
+    "System preset: NTSC - MPG",
+    "System preset: PAL - MPG",
+    "System preset: Full HD 1080i60 - MP4",
+    "System preset: Full HD 1080i50 - MP4",
+    "System preset: Gif (Animated)",
+    "System preset: Web: Flash Video",
+    "System preset: Full HD 1080i60 - XDCAM422",
+    "System preset: Full HD 1080i50 - XDCAM422",
+    "System preset: Webm 720p",
+    "System preset: Webm VP9 720p",
+]
+```
+
+_NOTE_: not the full list
+
+For the friendly names, these are essentially system presets, linked to a more friendly name in the appsettings file, under TranscoderMappings like this:
+
+``` json
+  "TimebasedIngest": {
+    "PipelineName": "dlcsspinup-timebased",
+    "TranscoderMappings": {
+      "Wellcome Standard MP4": "System preset: Web"
+    }
+  },
+```
+
+From this, the preset is used to kick off an ET `job` and then get pushed into buckets for output
+
+## Delivery Channel Proposal
+
+Firstly, as [discussed](https://github.com/dlcs/protagonist/issues/709), The `policyData` is an array of policies *which each link to a single system preset*, as opposed to combined values having multiple outputs.  These values are then looped through to create all the outputs that are required.  From an orchestrator perspective, nothing will have changed as these transcoder outputs will not change.
+
+Due to the above, the likely best place to put these values is a dictionary in appsettings, similar to how `TranscoderMappings` works.  This is because there are a limited number of presets we want to support (dev currently uses only 3), rather than all of them and that it is unlikely for these values to need to be changed regularly.  Additionally, `TranscoderMappings` should be deprecated, with a new setting created called `DeliveryChannelMappings`, which consists of key value pairs like the below:

```

Comment:
The only issue I can see with storing everything in appSettings is that if a single Protagonist deployment wants to add a new one it would need to replicate all of them, or use indexes to append (I _think_ this would work).

E.g. if `appSettings.json` in Protagonist has all system provided settings:

```json
  "TimebasedIngest": {
    "DeliveryChannelMappings": {
      "video-mp4-480p": "System preset: Generic 480p 16:9",
      "video-webm-720p": "System preset: Webm 720p(webm)",
      "audio-mp3-128k": "System preset: Audio MP3 - 128k(mp3)"
    }
  },
```

and I want to use those, but also add `"audio-flac-128": "my-custom-flac"`, then, assuming I'm using envvars in deployment, I'd need either to list all system ones, plus my own:

```
"TimebasedIngest__DeliveryChannelMappings__video-mp4-480p": "System preset: Generic 480p 16:9",
"TimebasedIngest__DeliveryChannelMappings__video-webm-720p": "System preset: Webm 720p(webm)",
"TimebasedIngest__DeliveryChannelMappings__audio-mp3-128k": "System preset: Audio MP3 - 128k(mp3)",
"TimebasedIngest__DeliveryChannelMappings__audio-flac-128": "my-custom-flac"
```

or (maybe - assuming this works) specify an index to add this to the end:

```
"TimebasedIngest__DeliveryChannelMappings__3__audio-flac-128": "my-custom-flac"
```

the issue with ^ is that I'd need to be aware of any new presets being added. What I'd really want to set is just my own value

```
"TimebasedIngest__DeliveryChannelMappings__audio-flac-128": "my-custom-flac"
```

Reply:
On the appsettings AV stuff,  I did some testing locally, with the following:

appsettings.json:
```
  "TimebasedIngest": {
    "DeliveryChannelMappings": {
      "TEST1": "System preset: Web",
      "TEST2": "some value"
    }
  }
```

appsettings.development.json:
```
  "TimebasedIngest": {
    "DeliveryChannelMappings": {
      "TEST2": "overloaded value",
      "TEST3": "someValue 3"
    }
  },
```
launchsettings.json:

```
"environmentVariables": {
        "TimebasedIngest__DeliveryChannelMappings__TEST3": "overloaded env var",
        "TimebasedIngest__DeliveryChannelMappings__TEST4": "new env var"
      }
```

final value was `{"TEST1":"System preset: Web","TEST2":"overloaded value","TEST3":"overloaded env var","TEST4":"new env var"}` - so I think this is ok